mmv1/products/gkeonprem/BareMetalCluster.yaml

```diff
@@ -0,0 +1,1003 @@
+update_url: "projects/{{project}}/locations/{{location}}/bareMetalClusters/{{name}}"
+delete_url: "projects/{{project}}/locations/{{location}}/bareMetalClusters/{{name}}?force=true"
+self_link: "projects/{{project}}/locations/{{location}}/bareMetalClusters/{{name}}"
+update_verb: :PATCH
+update_mask: true
+description: "A Google Bare Metal User Cluster."
+autogen_async: false
+id_format: "projects/{{project}}/locations/{{location}}/bareMetalClusters/{{name}}"
+import_format: ["projects/{{project}}/locations/{{location}}/bareMetalClusters/{{name}}"]
+taint_resource_on_failed_create: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "gkeonprem_bare_metal_cluster_basic"
+    min_version: beta
+    primary_resource_id: "cluster-basic"
+    vars:
+      name: "basic"
+    test_env_vars:
+      project: "fake-backend-360322"
+  - !ruby/object:Provider::Terraform::Examples
+    name: "gkeonprem_bare_metal_cluster_manuallb"
+    min_version: beta
+    primary_resource_id: "cluster-manuallb"
+    vars:
+      name: "manuallb"
+    test_env_vars:
+      project: "fake-backend-360322"
+  - !ruby/object:Provider::Terraform::Examples
+    name: "gkeonprem_bare_metal_cluster_bgplb"
+    min_version: beta
+    primary_resource_id: "cluster-bgplb"
+    vars:
+      name: "bgplb"
+    test_env_vars:
+      project: "fake-backend-360322"
+parameters:
+  - !ruby/object:Api::Type::String
+    name: "name"
+    description: The bare metal cluster name.
+    immutable: true
+    url_param_only: true
+    required: true
+  - !ruby/object:Api::Type::String
+    name: "location"
+    description: The location of the resource.
+    immutable: true
+    url_param_only: true
+    required: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: "adminClusterMembership"
+    required: true
+    immutable: true
+    description: |
+      The Admin Cluster this Bare Metal User Cluster belongs to.
+      This is the full resource name of the Admin Cluster's hub membership.
+    diff_suppress_func: ProjectNumberDiffSuppress
+  - !ruby/object:Api::Type::String
+    name: "description"
+    description: |
+      A human readable description of this Bare Metal User Cluster.
+  - !ruby/object:Api::Type::String
+    name: "bareMetalVersion"
+    required: true
+    description: |
+      A human readable description of this Bare Metal User Cluster.
+  - !ruby/object:Api::Type::KeyValuePairs
+    name: "annotations"
+    description: |
+      Annotations on the Bare Metal User Cluster.
+      This field has the same restrictions as Kubernetes annotations.
+      The total size of all keys and values combined is limited to 256k.
+      Key can have 2 segments: prefix (optional) and name (required),
+      separated by a slash (/).
+      Prefix must be a DNS subdomain.
+      Name must be 63 characters or less, begin and end with alphanumerics,
+      with dashes (-), underscores (_), dots (.), and alphanumerics between.
+    default_from_api: true
```

Comment:
Any reason to set this field default_from_api? Does the API return a default value back for an annotation?

Reply:
Yes, API will add some default annotations automatically on server side. We need to mark it as default_from_api to avoid unintentional override on this field during update. Thanks for pointing this out!